app/maptools/recordingmaptool.h

```diff
@@ -59,7 +59,7 @@ class Vertex
 
     bool operator==( const Vertex &other )
     {
-      return other.vertexId() == mVertexId && other.type() == mType;
+      return other.vertexId() == mVertexId && other.coordinates() == mCoordinates && other.type() == mType;
```

Comment:
Maybe better use `InputUtils::equals()` to compare coordinates to avoid rounding issues?